apps/meteor/tests/e2e/user-card-info-in-room.spec.ts

```diff
@@ -0,0 +1,56 @@
+import { Users } from './fixtures/userStates';
+import { HomeChannel } from './page-objects';
+import { createTargetChannel, deleteChannel } from './utils';
+import { expect, test } from './utils/test';
+
+test.use({ storageState: Users.admin.state });
+
+test.describe.parallel('Mention User Card', () => {
+	let poHomeChannel: HomeChannel;
+	let targetChannel: string;
+
+	test.beforeAll(async ({ api }) => {
+		targetChannel = await createTargetChannel(api, { members: [Users.user1.data.username] });
+
+		await api.post(`/chat.postMessage`, {
+			text: `Hello @${Users.user1.data.username} @${Users.user2.data.username}`,
+			channel: targetChannel,
+		});
+	});
+
+	test.beforeEach(async ({ page }) => {
+		poHomeChannel = new HomeChannel(page);
+
+		await page.goto('/home');
+	});
+
+	test.afterAll(({ api }) => deleteChannel(api, targetChannel));
+
+	test('should show correct userinfo actions for a member of the room', async ({ page }) => {
+		await poHomeChannel.sidenav.openChat(targetChannel);
+		const mentionSpan = page.locator(`span[title="Mentions user"][data-uid="${Users.user1.data.username}"]`);
+		await mentionSpan.click();
+
+		await expect(page.locator('button[title="Add to room"]')).not.toBeVisible();
+
+		await page.locator('div[aria-label="User card actions"] button[title="More"]').click();
+
+		await expect(page.locator('label[data-key="Remove from room"]')).toBeVisible();
+		await expect(page.locator('label[data-key="Set as leader"]')).toBeVisible();
+		await expect(page.locator('label[data-key="Set as moderator"]')).toBeVisible();
```

Comment:
For this to be true, the user must be the room owner (or at least they must have the necessary permission), right? 🤔  I belive being a member only doesn't mean you can add leaders, removing people from the room, etc...

If so, we need to put this in the test title `should show correct userinfo actions for the room owner`, and create another test cases for when the user is not the owner/don't have permissions for removing from room, set as leader, etc.

Reply:
Yes, you are right :)
Moving forward I have added tests for both when viewed by room owner and when viewed by normal member.
I have also split the test test files into two  in order to prevent the timeout issue that results in flaky tests.

Also, the action button can be visible in the card or it could be hidden under the more button. So, in order to prevent flaky tests that might arise in future, I have added check for both the cases.